schema/validator.go

```diff
@@ -28,6 +30,12 @@ import (
 // and implements validation against a JSON schema.
 type Validator string
 
+type validateDescendantsFunc func(r io.Reader) error
+
+var mapValidateDescendants = map[Validator]validateDescendantsFunc{
+	MediaTypeManifest: validateManifestDescendants,
```

Comment:
I agree that you wouldn't want all of recursive validation, all recursive validation code in image-spec, and CAS-engine code in image-tools.  My personal preference there is to [do non-recursive JSON Schema validation here and to handle recursive validation and the CAS-engine in image-tools][1].  @stevvooe [seems to][2] [prefer][3] keeping more than just JSON Schema validation in image-spec, but @philips has [moved a fair bit of stuff to image-tools][4], so I'm not sure there's a clear maintainer position on this yet.  Once there is a clear position, it should be fairly straightforward to put this logic in the approved location.

[1]: https://github.com/opencontainers/image-spec/pull/337#issuecomment-248816631
[2]: https://github.com/opencontainers/image-spec/pull/337#issuecomment-249673277
[3]: https://github.com/opencontainers/image-tools/pull/10#issuecomment-249670609
[4]: https://github.com/opencontainers/image-spec/pull/337#issuecomment-249670746